### ISSUE TITLE
Pokemon are not evolving if caught at level above when they would normally evolve

### ISSUE THREAD

**Issue by jrsukinik** [Unlucky-Life/ankimon#82]
Hello! I am new to the add-on and have been loving it. However, i recently caught a number of pokemon above lvl 40 (for example teddiursa), which are not evolving after being leveled up. Do you have an idea of what is going on?

**Comment by jrsukinik** [comment]
So far teddiursa, although tomorrow I will know more as I am close to leveling up a few.

**Comment by jrsukinik** [comment]
azurill as well now

**Comment by Unlucky-Life** [comment]
@jrsukinik i found it. This issue is that right now pokemon that need certain conditions to evolve is not implemented. Right now you can only evolve pokemon with items needed for evolution aswell as by  leveling your pokemon.

**Comment by jrsukinik** [comment]
okay, so even though teddiursa evolves into ursaring at 32 without any special conditions, since the final evolution has special requirements, it will not evolve at all?

**Comment by Unlucky-Life** [comment]
@jrsukinik Teddiursa evolves at level 30 https://bulbapedia.bulbagarden.net/wiki/Teddiursa_(Pok%C3%A9mon)
![image](https://github.com/Unlucky-Life/ankimon/assets/77027147/58e1ce59-d57f-4208-b9fe-a8f165857c79)
But actually it should have evoled at level 30. Quite weird.

**Comment by Unlucky-Life** [comment]
@jrsukinik 
![image](https://github.com/Unlucky-Life/ankimon/assets/77027147/cd351026-b94e-440f-8f2c-2d92b57a7e33)
azurill cant evolve right now because it need the friendship level and I didnt know how i could integrate that yet. So yea azurill wont evolve

**Comment by jrsukinik** [comment]
noticed the same with machop too. Leveled up from 36-37 with no evolution. Maybe pokemon with special requirements for any evolution are having difficulty. I will try out a pokemon that has no special requirements at all

**Comment by Unlucky-Life** [comment]
@jrsukinik Yea exactly - if a pokemon needs certain requirements aswell it wont evolve. (Didnt integrate this yet because it very time intensive to code this)

**Comment by jrsukinik** [comment]
Out of curiosity, is it possible to add >= to the level for evolution for a pokemon? Perhaps that would solve it, saying \, for example, "evolevel": >=28?

**Comment by Unlucky-Life** [comment]
@jrsukinik Yes that should be possible ! I will also be adding a everstone option for people who dont want to evolve their pokemon !

**Comment by lqvu2205** [comment]
I have the same issue. My Rayquaza met the requirement for Mega Evolution but nothing happened. Mega Evolution isn't available yet, is it? If so, can you add that feature in the future? Thank you!
![image](https://github.com/Unlucky-Life/ankimon/assets/129834552/11e5b5de-3d22-4861-96a7-72d9d32e9e2c)
![image](https://github.com/Unlucky-Life/ankimon/assets/129834552/7285720f-3730-4547-a824-38c4ed3d826c)

**Comment by Unlucky-Life** [comment]
@lqvu2205 There are alot of feature requests, but I'm tryign to implement this aswell :)
